src/gui/dialog/SettingsDialog.h

```diff
@@ -11,13 +11,18 @@
 
 #pragma once
 
+#ifdef ENABLE_AUDIO
 #include <util/audio/DeviceInfo.h>
+#endif
 #include <control/Control.h>
 #include "control/settings/Settings.h"
 #include "gui/GladeGui.h"
 #include "DeviceClassConfigGui.h"
 
 class ButtonConfigGui;
+#ifndef ENABLE_AUDIO
+class DeviceInfo;
+#endif
```

Comment:
move include to source file and leave `class DeviceInfo`, remove preprozessor stmts